### PR TITLE
Enable OIDC auth

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by akatsoulas** [comment]
Please make sure that you are not removing any functionality. Eg if the user authenticates through the OIDC flow then we need to make sure that the user is a vouched mozillian

**Comment by johngian** [comment]
@akatsoulas  PR updated

* Pinned ``mozilla_django_oidc`` version and hashes
* Added custom browserid functionality
    * ``login_failure``
    * check for vouched status

File: moderator/settings.py [hunk 1]
```diff
@@ -192,11 +180,35 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
...
+    import hashlib
+
+    try:
+        from django.utils.encoding import smart_bytes
+    except ImportError:
+        from django.utils.encoding import smart_str as smart_bytes
+
+    return base64.urlsafe_b64encode(
+        hashlib.sha1(smart_bytes(email)).digest()
+    ).rstrip(b'=')
```
Comment (akatsoulas) [inline moderator/settings.py]: Why don't you use the `username_algo` provided by the lib?

Comment (johngian) [inline moderator/settings.py]: Because the project was older than 1.8 and has username length less than 30.

**Comment by akatsoulas** [comment]
r+

**Pull request merged**